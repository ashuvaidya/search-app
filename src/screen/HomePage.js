import { memo, useState } from "react";
import propTypes from "prop-types";
import { Header } from "./Header";
import { ResultCard } from "../commonComponent/ResultCard";
import { search } from "../api/search";

const HomePage = memo(() => {
  const [isLoading, setIsLoading] = useState(false);
  const [searchResult, setSearchResult] = useState(null);
  const [searchString, setSearchString] = useState("");

  const handleSearch = async () => {
    if (searchString.length > 0) {
      const results = await search(searchString, setIsLoading);
      setSearchResult(results);
    }
  };

  return (
    <>
      <Header />
      <div className="container">
        <div className="search-section">
          <input
            data-testid="input-search"
            type="search"
            aria-label="search"
            className="search-box"
            value={searchString}
            onChange={(e) => setSearchString(e.target.value)}
          />
          <button
            name="searchBtn"
            data-testid="searchBtn"
            className="search-button"
            onClick={() => handleSearch()}
          >
            Search
          </button>
        </div>
        <div className="result-section">
          {isLoading ? (
            <div> Fetching Details. Please wait...</div>
          ) : (
            <>
              {searchResult?.length > 0 && <h2>Results: </h2>}
              {searchResult?.map((data, index) => {
                return (
                  <ResultCard
                    data={data}
                    key={`result-${index}`}
                    id={`result-${index}`}
                  />
                );
              })}
            </>
          )}
        </div>
      </div>
    </>
  );
});

HomePage.propTypes = {
  search: propTypes.func,
  isLoading: propTypes.bool,
};

export { HomePage };
