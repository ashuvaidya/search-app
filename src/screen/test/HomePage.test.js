import { render, fireEvent } from "@testing-library/react";
import { HomePage } from "../HomePage";

const setup = () => {
  const { container, getByTestId } = render(<HomePage />);
  return { container, getByTestId };
};

describe("HomePage", () => {
  it("renders homepage component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it("click on search", () => {
    const { container, getByTestId } = setup();
    const searchInput = getByTestId("input-search");
    fireEvent.change(searchInput, { target: { value: "test" } });
    expect(searchInput.value).toBe('test');
    const searchBtn = getByTestId("searchBtn");
    fireEvent.click(searchBtn);
    expect(container).toMatchSnapshot();
  });
});
