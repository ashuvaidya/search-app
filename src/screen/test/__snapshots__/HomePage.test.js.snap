// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage click on search 1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      Search Engine
    </h1>
  </div>
  <div
    class="container"
  >
    <div
      class="search-section"
    >
      <input
        aria-label="search"
        class="search-box"
        data-testid="input-search"
        type="search"
        value="test"
      />
      <button
        class="search-button"
        data-testid="searchBtn"
        name="searchBtn"
      >
        Search
      </button>
    </div>
    <div
      class="result-section"
    >
      <div>
         Fetching Details. Please wait...
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage renders homepage component 1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      Search Engine
    </h1>
  </div>
  <div
    class="container"
  >
    <div
      class="search-section"
    >
      <input
        aria-label="search"
        class="search-box"
        data-testid="input-search"
        type="search"
        value=""
      />
      <button
        class="search-button"
        data-testid="searchBtn"
        name="searchBtn"
      >
        Search
      </button>
    </div>
    <div
      class="result-section"
    />
  </div>
</div>
`;
